### PR TITLE
[PANGOO-2581][BpkNavigationTabGroup]Add Extra Props in BPK NavTabGroup

### PR DESCRIPTION
**Pull request by kerrie625**
When using BPK Nav tabs in GC header, we found MouseEvent need pass  from  BPK NavTabs

There is click event function in GC header to handle all click event in Header .

And also need tab id and nav id pros in BPK NavTabs.  For data analytics, nav tabs also need data analytics props as optional.  

- [x] Ensure the PR title includes the name of the component you are changing so it's clear in the release notes for consumers of the changes in the version e.g `[KOA-123][BpkButton] Updating the colour`
- [x] `README.md` (If you have created a new component)
- [x] Component `README.md`
- [x] Tests
- [x] Accessibility tests
    - The following checks were performed:
        - [x] Ability to navigate using a [keyboard only](https://webaim.org/techniques/keyboard/)
        - [x] Zoom functionality ([Deque University explanation](https://dequeuniversity.com/checklists/web/text)):
            - [x] The page SHOULD be functional AND readable when only the text is magnified to 200% of its initial size
            - [x] Pages must reflow as zoom increases up to 400% so that content continues to be presented in only one column i.e. Content MUST NOT require scrolling in two directions (both vertically and horizontally)
        - [x] Ability to navigate using a [screen reader only](https://webaim.org/articles/screenreader_testing/)
- [x] Storybook examples created/updated
- [ ] For breaking changes or deprecating components/properties, migration guides added to the description of the PR. If the guide has large changes, consider creating a new Markdown page inside the component's docs folder and link it here

### CODE REVIEW

File: packages/bpk-component-navigation-tab-group/src/BpkNavigationTabGroup.tsx [hunk 1]
```diff
@@ -34,17 +34,21 @@ export type NavigationTabGroupTypes =
...
 };
 export type Props = {
+  id: string;
   tabs: TabItem[];
   type?: NavigationTabGroupTypes;
   /*
    * Index parameter to track which is clicked
    */
-  onItemClick: (tab: TabItem, index: number) => void;
+  onItemClick: (e: MouseEvent<HTMLButtonElement | HTMLAnchorElement>,tab: TabItem, index: number) => void;
```
Comment (kerrie625) [inline packages/bpk-component-navigation-tab-group/src/BpkNavigationTabGroup.tsx:51]: https://github.skyscannertools.net/skyscanner/global-components/blob/3bd5807e96de27f95312336845d14b573b6cf24e/src/header/utils/handleClickAndTrack.ts#L24-45

File: packages/bpk-component-navigation-tab-group/src/BpkNavigationTabGroup.tsx [hunk 2]
```diff
@@ -64,22 +68,29 @@ const TabWrap = ({ children, onClick, selected, tab, type }: TabWrapProps) => {
     selected && `bpk-navigation-tab-wrap--${type}-selected`,
   );
 
+  const tabProps = {
+    ...(tab.dataCy && { 'data-cy': tab.dataCy }),
+    ...(tab.dataAnalyticsName && { 'data-analytics-name': tab.dataAnalyticsName }),
```
Comment (kerrie625) [inline packages/bpk-component-navigation-tab-group/src/BpkNavigationTabGroup.tsx:73]: Props 'data-cy' and 'data-analytics-name' has been used in GC header. To order to not impact header , add optional props in Nav tabs
https://github.skyscannertools.net/skyscanner/global-components/blob/e6dbbff3baa33e1a4a380dc2b27c10912cf8397e/src/header/HeaderTab/HeaderTab.tsx#L90

File: packages/bpk-component-navigation-tab-group/src/BpkNavigationTabGroup.tsx (hunk 1 above)
Comment (kerrie625) [inline packages/bpk-component-navigation-tab-group/src/BpkNavigationTabGroup.tsx:51]: In GC header all click event and track be handled by handleClickAndTrack function that need MouseEvent ,so added args there
 https://github.skyscannertools.net/skyscanner/global-components/blob/e6dbbff3baa33e1a4a380dc2b27c10912cf8397e/src/header/utils/handleClickAndTrack.ts#L24

File: packages/bpk-component-navigation-tab-group/src/BpkNavigationTabGroup.tsx [hunk 3]
```diff
@@ -34,17 +34,21 @@ export type NavigationTabGroupTypes =
   (typeof NAVIGATION_TAB_GROUP_TYPES)[keyof typeof NAVIGATION_TAB_GROUP_TYPES];
 
 type TabItem = {
+  id: string;
   text: string;
   icon?: FunctionComponent<any> | null;
   href?: string;
+  dataCy?: string;
```
Comment (olliecurtis) [inline packages/bpk-component-navigation-tab-group/src/BpkNavigationTabGroup.tsx:41]: Could you change the name from `cy` to something more descriptive as this is slightly confusing as to what cy means?

Comment (kerrie625) [reply packages/bpk-component-navigation-tab-group/src/BpkNavigationTabGroup.tsx]: Change dataCy to dataCypress as it used for cypress test .[ec83984](https://github.com/Skyscanner/backpack/pull/3607/commits/ec83984a5d56b0ad45bdf755a5cfc38615aecf4d)

File: packages/bpk-component-navigation-tab-group/src/BpkNavigationTabGroup.tsx (hunk 2 above)
Comment (kerrie625) [reply packages/bpk-component-navigation-tab-group/src/BpkNavigationTabGroup.tsx:73]: And also some project are using data-cy to verify header loaded.

File: packages/bpk-component-navigation-tab-group/src/BpkNavigationTabGroup.tsx (hunk 3 above)
Comment (kerrie625) [reply packages/bpk-component-navigation-tab-group/src/BpkNavigationTabGroup.tsx]: Thanks for your help. I have resolved this props issue .Updated it in [7b6a1d9](https://github.com/Skyscanner/backpack/pull/3607/commits/7b6a1d94188ebf04c662a5c5b43524a9835c602a)

**Pull request merged**